app/src/Model/Table/ArticlesTable.php

```diff
@@ -0,0 +1,63 @@
+<?php
+declare(strict_types=1);
+
+// src/Model/Table/ArticlesTable.php
+namespace App\Model\Table;
+
+// the EventInterface class
+use Cake\Event\EventInterface;
+use Cake\ORM\Table;
+// the Text class
+use Cake\Utility\Text;
+// the Validator class
+use Cake\Validation\Validator;
+
+class ArticlesTable extends Table
+{
+    /**
+     * Initialize the table and add timestamp behavior.
+     *
+     * @param array $config Additional table configuration.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        $this->addBehavior('Timestamp');
+    }
+
+    /**
+     * Before saving an entity, generate a slug if it doesn't exist.
+     *
+     * @param \Cake\Event\EventInterface $event Save event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity being saved.
+     * @param array $options Additional options.
+     * @return void
+     */
+    public function beforeSave(EventInterface $event, $entity, $options)
+    {
+        if ($entity->isNew() && !$entity->slug) {
+            $sluggedTitle = Text::slug($entity->title);
+            // trim slug to maximum length defined in schema
+            $entity->slug = substr($sluggedTitle, 0, 191);
+        }
+    }
+
+    /**
+     * Define default validation rules for the articles entity.
+     *
+     * @param \Cake\Validation\Validator $validator Validator to be used.
+     * @return \Cake\Validation\Validator The configured validator.
+     */
+    public function validationDefault(Validator $validator): Validator
+    {
+        $validator
+            ->notEmptyString('title')
+            ->minLength('title', 10)
+            ->maxLength('title', 255)
+
+            ->notEmptyString('body')
+            ->minLength('body', 10);
+
+        return $validator;
+    }
```

Comment:
The CakePHP tutorial defined the field `slug` as unique. Refer https://github.com/ArielGalvez/tutorial-cakephp/blob/3726ad5628386cc4d3d485ac59d3aa768732e3ea/app/config/Migrations/20231025051828_CreateArticles.php#L19

How do we ensure the slug is unique?

Reply:
Additionally, please refer this topic https://book.cakephp.org/4/en/orm/validation.html#validation-vs-application-rules

There're differences between validation & application rules.

Reply:
Regarding my previous comment related to the `bake` command. You may try to run the bake command to generate the model file as well. Please check & identify the differences between the copied code from the tutorial & the generated one.

Reply:
Refer back to my comment https://github.com/ArielGalvez/tutorial-cakephp/pull/2#discussion_r1377434146

Do you have any idea to ensure the slug is unique?

Reply:
Yes, we can add a validation to the title in the form